File: Lista de Exercícios/Go/14. Singleton/Singleton.go
package main

import (
	"fmt"
	"sync"
)

type Configuracao struct {
	configuracoes map[string]string
}

var (
	instancia *Configuracao
	once      sync.Once
)

func GetConfiguracao() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{
			configuracoes: make(map[string]string),
		}
	})
	return instancia
}

func main() {

}

File: Lista de Exercícios/Java/14. Singleton/Singleton.java
class Configuracao {
    private static Configuracao instancia;
    private java.util.Map<String, String> configuracoes;

    private Configuracao() {
        configuracoes = new java.util.HashMap<>();
    }

    public static Configuracao getInstancia() {
        if (instancia == null) {
            instancia = new Configuracao();
        }
        return instancia;
    }

    public void adicionarConfiguracao(String chave, String valor) {
        configuracoes.put(chave, valor);
    }

    public String obterConfiguracao(String chave) {
        return configuracoes.get(chave);
    }
}
